cpu/stm32/Makefile.include

```diff
@@ -1,7 +1,12 @@
 CFLAGS += -DCPU_FAM_STM32$(call uppercase_and_underscore,$(CPU_FAM))
 
-# For stm32 cpu's we use the generic stm32.ld linker script
-LINKER_SCRIPT ?= stm32.ld
+# For stm32 cpu's we use the generic stm32.ld linker script,
+# stm32u5 has it's own linker script
+ifeq (u5,$(CPU_FAM))
+  LINKER_SCRIPT ?= stm32u5.ld
+else
+  LINKER_SCRIPT ?= stm32.ld
```

Comment:
Oh I didn't realize we already have a custom linker script for stm32 - then let's use this instead

```
diff --git a/cpu/stm32/Makefile.include b/cpu/stm32/Makefile.include
index 884890ccc1..406f58a34c 100644
--- a/cpu/stm32/Makefile.include
+++ b/cpu/stm32/Makefile.include
@@ -61,6 +61,9 @@ info-stm32:
 ifneq (,$(CCMRAM_LEN))
   LINKFLAGS += $(LINKFLAGPREFIX)--defsym=_ccmram_length=$(CCMRAM_LEN)
 endif
+ifneq (,$(SRAM4_LEN))
+  LINKFLAGS += $(LINKFLAGPREFIX)--defsym=_sram4_length=$(SRAM4_LEN)
+endif
 
 VECTORS_O ?= $(BINDIR)/stm32_vectors/$(CPU_LINE).o
 VECTORS_FILE = $(RIOTCPU)/stm32/vectors/$(CPU_LINE).c
diff --git a/cpu/stm32/ldscripts/stm32.ld b/cpu/stm32/ldscripts/stm32.ld
index fb50fcf119..3e429d879d 100644
--- a/cpu/stm32/ldscripts/stm32.ld
+++ b/cpu/stm32/ldscripts/stm32.ld
@@ -19,10 +19,12 @@
  */
 
 ccmram_length = DEFINED( ccmram_len ) ? ccmram_len : 0x0 ;
+sram4_length = DEFINED( sram4_length ) ? sram4_length : 0x0 ;
 
 MEMORY
 {
     ccmram  : ORIGIN = 0x10000000, LENGTH = ccmram_length
+    sram4   : ORIGIN = 0x28000000, LENGTH = sram4_length
 }
 
 INCLUDE cortexm.ld
```

Reply:
Ah but why remove

```C
#define NUM_HEAPS   2
```
from `cpu_conf.h `?

Reply:
I don't really know why it's required, I thought it was useless.

Reply:
It's used by [`syscalls.c`](https://github.com/RIOT-OS/RIOT/blob/master/sys/newlib_syscalls_default/syscalls.c#L106) to provide non-continuous memory sections as additional heap used by `malloc`